sdk/core/Azure.Core/src/Shared/ClientDiagnostics.cs

```diff
@@ -34,7 +41,28 @@ public DiagnosticScope CreateScope(string name)
             {
                 return default;
             }
-            return new DiagnosticScope(name, _source);
+            var scope = new DiagnosticScope(name, _source);
+
+            if (_resourceProviderNamespace != null)
+            {
+                scope.AddAttribute("az.namespace", _resourceProviderNamespace);
+            }
+            return scope;
+        }
+
+        internal static string? GetResourceProviderNamespace(Assembly assembly)
+        {
+            foreach (var customAttribute in assembly.GetCustomAttributes(true))
```

Comment:
Isn't `GetCustomAttributes<T>` in netstandard2.0? IIRC, it should be faster because it's either handled by the runtime, or at least may be NGEN'd than looping through types ourselves.

Reply:
Not referencing type directly because it would be a "new" type for every package.